### PR TITLE
Engine param

### PR DESCRIPTION
**Pull request by sadrasabouri**
#### Reference Issues/PRs
#145 
#### What does this implement/fix? Explain your changes.
This PR will add `engine` parameter to the PyRGG menu. Until now we only have one engine (Pyrgg engine) but new engines will be proposed using this architecture.

I tried to develop it as cohesive as I can. Using `engine` parameter from `result_dict` we can control parameters needed for each generation method.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- generation engine menu
```
Comment (sepandhaghighi) [inline CHANGELOG.md:9]: Generation engine menu

**Review by sepandhaghighi — CHANGES_REQUESTED** [review]
@sadrasabouri Thanks for your efforts 💯 

1. To make it possible to work with various engines dynamically, it is necessary to update both `get_input` and `_update_using_second_menu` functions.
2. I suggest renaming `MENU_ITEMS2` to a more appropriate name like `PYRGG_ENGINE_PARAMS`.
3. I agree with you regarding the bug in the order of `MENU_ITEMS1`. To resolve this issue, we need to convert it into the `MENU_ITEMS2` format.

**Comment by sadrasabouri** [comment]
> 

1. Done (although I think we can use a class or at least an Enum for engines in future, too)
2. Done
3. That's a great approach. It's better to deal with in a separate PR.

**Review by sepandhaghighi — APPROVED** [review]
🚀

**Pull request merged**